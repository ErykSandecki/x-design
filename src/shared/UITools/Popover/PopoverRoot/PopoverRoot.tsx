import { FC, ReactNode } from 'react';

// core
import { PopoverRootProvider } from './core/PopoverRootProvider';

// types
import { TContext } from './core/types';

export type TPopoverRootProps = TContext & {
  children: ReactNode;
};

export const PopoverRoot: FC<TPopoverRootProps> = ({ children, selected, setSelected }) => (
  <PopoverRootProvider selected={selected} setSelected={setSelected}>
    {children}
  </PopoverRootProvider>
);

export default PopoverRoot;
