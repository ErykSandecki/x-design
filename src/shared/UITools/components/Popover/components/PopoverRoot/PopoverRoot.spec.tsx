import { noop } from 'lodash';
import { render } from '@testing-library/react';

// components
import PopoverRoot from './PopoverRoot';

describe('PopoverRoot snapshots', () => {
  it('should render PopoverRoot', () => {
    // before
    const { asFragment } = render(
      <PopoverRoot setSelected={noop}>children</PopoverRoot>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
