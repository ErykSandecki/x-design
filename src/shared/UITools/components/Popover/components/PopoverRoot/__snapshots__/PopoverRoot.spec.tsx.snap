// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverRoot snapshots should render PopoverRoot 1`] = `
<DocumentFragment>
  children
</DocumentFragment>
`;
